### PR TITLE
Move `Plugin.Execute` to framework -- I KNOW, THIS IS A HUGE PR, DONT HATE ME! 

### PR DESCRIPTION
**Pull request by klesh**
# Summary

Move `Plugin.Execute` to framework
Decopule global variables

### Key Points

- [x] This is a breaking change
- [ ] New or existing documentation is updated

### CODE REVIEW

File: plugins/core/plugin_migration.go [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package core
+
+// TODO: Implement this interface to register database migration-scripts for your plugin
+type PlugibMigrations interface {
```
Comment (warren830) [inline plugins/core/plugin_migration.go:4]: rename to pluginMigration

File: .env.example [hunk 2]
```diff
@@ -15,6 +17,12 @@ MODE=release
 NOTIFICATION_ENDPOINT=
 NOTIFICATION_SECRET=
 
+API_TIMEOUT=10
+API_RETRY=3
+API_REQUESTS_PER_HOUR=10000
```
Comment (warren830) [inline .env.example:22]: I'm thinking about adding a retry interval, If we retry without an interval, it might cause rate limit exceed

**Review by warren830 — APPROVED** [review]
LGTM, fantastic design, great way to manage sub tasks. And for the future contributors, it will be much easier to build a new plugin

File: .env.example (hunk 2 above)
Comment (klesh) [reply .env.example:22]: retry was moved to `ApiAsyncClient`, and using `GetAsync` method, it was under control of rate limit

**Review by mindlesscloud — APPROVED** [review]

**Pull request merged**